synapse/models/infotech.py

```diff
@@ -1076,6 +1095,113 @@ def getModelDefs(self):
+                    }),
+                    ('type', ('it:dev:repotype', {}), {
+                        'doc': 'The type of the version control system used.',
+                        'ex': 'svn'
+                    }),
+                    ('submodules', ('array', {'type': 'it:dev:repo:commit'}), {
+                        'doc': "An array of other repos that this repo has as submodules, pinned at specific commits.",
+                    }),
+                )),
+
+                ('it:dev:repo:commit', {}, (
+                    ('repo', ('str', {}), {
+                        'doc': 'The repository the commit lives in.',
+                    }),
+                    ('parents', ('array', {'type': 'it:dev:repo:commit'}), {
+                        'doc': 'The commit or commits this commit is immediately based on.',
+                    }),
+                    ('branch', ('str', {}), {
+                        'doc': 'The name of the branch the commit was made to.',
+                    }),
+                    ('mesg', ('str', {}), {
+                        'disp': {'hint': 'text'},
+                        'doc': 'The commit message describing the changes in the commit.',
+                    }),
+                    ('id', ('str', {}), {
+                        'doc': 'The version control system specific commit identifier.',
+                    }),
+                    ('created', ('time', {}), {
+                        'doc': 'When the commit was made.',
+                    }),
+                )),
+
+                ('it:dev:repo:diff', {}, (
+                    ('commit', ('it:dev:repo:commit', {}), {
+                        'doc': 'The commit that produced this diff.',
+                    }),
+                    ('file', ('file:bytes', {}), {
+                        'doc': 'The file after the commit has been applied',
+                    }),
+                    ('path', ('file:path', {}), {
+                        'doc': 'The path to the file in the repo that the diff is being applied to.',
+                    }),
+                )),
+
+                ('it:dev:repo:issue', {}, (
+                    ('repo', ('it:dev:repo', {}), {
+                        'doc': 'The repo where the issue was logged.',
+                    }),
+                    ('title', ('str', {'lower': True, 'strip': True}), {
+                        'doc': 'The title of the issue.'
+                    }),
+                    ('desc', ('str', {}), {
+                        'disp': {'hint': 'text'},
+                        'doc': 'The text describing the issue.'
+                    }),
+                    ('created', ('time', {}), {
+                        'doc': 'The time the issues created.',
+                    }),
+                )),
+
+                ('it:dev:repo:comment', {}, (
+                    ('repo', ('it:dev:repo', {}), {
+                        'doc': 'The repo the comment was added to.',
+                    }),
```

Comment:
Do we perhaps need a `diff` ( or maybe `commit` ? ) field here? Does that render `repo` redundant? ( topics for discussion :) )

Reply:
I think `commit` makes the most sense? `diff` doesn't really add anything that isn't already hanging off of `comment`.

It does raise the specter of whether `it:dev:repo:comment` should be split into `issue:comment` and `diff:comment`. I initially opted against it since there was a far amount of overlap, but now I'm wondering if it'd be worth it. Another topic for discussion.